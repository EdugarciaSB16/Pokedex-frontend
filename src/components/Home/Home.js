/**
 * Home Page
 *
 */
// Dependencias
import React from 'react';
/**
 * Contenedor de la página principal
 *
 */
export const Home = () => {
    return (
        <div className="Home">
            <h1>Welcome to the pokédex</h1>

            <div>
                <a href="/pokemons" className="btn">
                    Start
                </a>
            </div>
        </div>
    );
};
