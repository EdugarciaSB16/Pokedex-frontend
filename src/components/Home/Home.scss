.Home {
    height: 100vh;
    background-image: url('https://cdn.dribbble.com/users/1771704/screenshots/5891354/shiny-charizard.gif');
    background-position: center;
    background-repeat: no-repeat;
    background-size: 700px;
    background-color: #252628;

    h1 {
        color: white;
        text-align: center;
        padding-top: 100px;
        font-family: 'Dancing Script', cursive;
        font-size: 62px;
    }

    a {
        position: absolute;
        top: 80%;
        left: 50%;
        transform: translate(-50%, -50%);
        background-color: #d0003d;
        color: white;
        width: 130px;
        font-size: 22px;
    }
}

@media (max-width: 768px) {
    .Home {
        background-size: 450px;
    }
}

@media (max-width: 468px) {
    .Home {
        background-size: 350px;
        h1 {
            font-size: 40px;
        }
    }
}
