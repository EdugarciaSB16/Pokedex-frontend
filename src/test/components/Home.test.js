/**
 * Test al componente Home
 * 
 */
// Dependencias
import React from 'react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow, configure } from 'enzyme';
// Componente
import { Home } from '../../components/Home/Home';

configure({ adapter: new Adapter() });

describe('Pruebas en el componente <Home />', () => {
    const wrapper = shallow(<Home />);
    const titleHome = 'Welcome to the pokÃ©dex';

    test('Debe de renderizar componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de renderizar el titulo principal', () => {
        const title = wrapper.find('h1');
        expect(title.text().trim()).toBe(titleHome);
    });

});
