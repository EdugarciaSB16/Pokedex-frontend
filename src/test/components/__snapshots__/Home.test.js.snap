// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <Home /> Debe de renderizar componente correctamente 1`] = `
<div
  className="Home"
>
  <h1>
    Welcome to the pok√©dex
  </h1>
  <div>
    <a
      className="btn"
      href="/pokemons"
    >
      Start
    </a>
  </div>
</div>
`;
